engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/EngineJobExecutionActor.scala

```diff
@@ -112,10 +113,32 @@ class EngineJobExecutionActor(jobKey: BackendJobDescriptorKey,
       writeToMetadata(Map(callCachingReadResultMetadataKey -> "Cache Miss"))
       log.info(s"Cache miss for job ${jobDescriptor.key.call.fullyQualifiedName}, index ${jobDescriptor.key.index}")
       runJob(jobDescriptor, bjeaProps)
-    case Event(CacheHit(cacheResultId), JobDescriptorData(jobDescriptor, _)) =>
-      writeToMetadata(Map(callCachingReadResultMetadataKey -> s"Cache Hit (from result ID $cacheResultId)"))
-      log.info(s"Cache hit for job ${jobDescriptor.key.call.fullyQualifiedName}, index ${jobDescriptor.key.index}! Copying cache result $cacheResultId")
-      lookupCachedResult(jobDescriptor, cacheResultId)
+    case Event(hit: CacheHit, JobDescriptorData(jobDescriptor, bjeaProps)) =>
+      writeToMetadata(Map(callCachingReadResultMetadataKey -> s"Cache Hit (from result ID ${hit.cacheResultId})"))
+      log.info(s"Cache hit for job ${jobDescriptor.key.call.fullyQualifiedName}, index ${jobDescriptor.key.index}! Copying cache result ${hit.cacheResultId}")
+      lookupCacheResult(jobDescriptor, bjeaProps, jobKey.call.task.outputs, hit)
+  }
+
+  // When FetchingCachedOutputsFromDatabase, the FSM should have EJEAJobDescriptorData
+  when(FetchingCachedOutputsFromDatabase) {
+    case Event(CachedOutputLookupSucceeded(cachedJobOutputs, cacheHit), JobDescriptorData(jobDescriptor, bjeaProps)) =>
+      copyCacheHit(cacheHit, jobDescriptor, bjeaProps, cachedJobOutputs)
+    case Event(CachedOutputLookupFailed(metaInfoId, error), JobDescriptorData(jobDescriptor, bjeaProps)) =>
+      log.info(s"Can't make a copy of the cached job outputs for $jobTag due to $error. Running job.")
+      runJob(jobDescriptor, bjeaProps)
+    case Event(_: CallCacheHashes, _) =>
+      // This situation might be possible - we might have hashes which are not required for determining cache hit
+      // (e.g. the memory runtime attribute) which are being reported after cache hit determination.
+      log.info(s"Getting these hashes when cache hit already found. Waiting for CachedJobOutputs")
```

Comment:
And make this a more troublesome error: `log.error("Hashes received while waiting for call-cache database read: This job will never complete. Much sad.")`